### ISSUE TITLE
TOC: adding "depth control"

### ISSUE THREAD

**Issue by scambier** [careteditor/caret#344]
Before recently making the switch to caret, I was mainly using Atom to write markdown, with this TOC plugin : https://atom.io/packages/markdown-toc

It has a nice feature that allows you to control the depth of it (i.e generate a TOC with titles from h2 to h4). I find it particularly useful because I usually write a single h1 title that is the document's title, and all my main sections are h2. Because of this, it's not desirable to have the h1 in my TOC.

Would it be possible to have a similar feature in caret? Something like `[[TOC from:2 to:4]]`?

**Comment by borekb** [comment]
My documents usually have the same structure so would use this as well.

Related: Convert [[toc]] to full Markdown #255

**Comment by astoilkov** [comment]
Yes. Definitely. What is the syntax in markdown-toc in Atom? Do you think the syntax there is convenient or needs improvement?

**Comment by scambier** [comment]
The plugin adds these commands in the CMD+Shift+P:

- Markdown Toc: Create
- Markdown Toc: Update
- Markdown Toc: Delete

It generates the TOC in markdown with some HTML comment around it:

```
<!-- TOC depthFrom:1 depthTo:6 withLinks:1 updateOnSave:1 orderedList:0 -->

(TOC generated in markdown appears here)

<!-- /TOC -->
```
The "withLinks" options simply adds an internal link to each title. It's neat but does not work (IIRC) when you want to export in pdf with another plugin.

> Do you think the syntax there is convenient or needs improvement?

I find it convenient as it is, it has all the options you need to have.
